Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TeamsController.cs

```diff
@@ -863,5 +974,96 @@ private async Task<List<ShiftsIntegResponse>> ProcessSwapShiftRequestApprovalAsy
+            List<ShiftsIntegResponse> responseModelList = new List<ShiftsIntegResponse>();
+
+            var openShiftRequest = this.GetOpenShiftRequest(jsonModel, approved);
+            var shift = this.GetShift(jsonModel, approved);
+            var openShift = this.GetOpenShift(jsonModel, approved);
+
+            bool success = false;
+
+            try
+            {
+                var openShiftRequestMapping = await this.openShiftRequestMappingEntityProvider.
+                    GetOpenShiftRequestMappingEntityByOpenShiftRequestIdAsync(openShiftRequest.OpenShiftId, openShiftRequest.Id).ConfigureAwait(false);
+                var kronosReqId = openShiftRequestMapping.KronosOpenShiftRequestId;
+                var kronosUserId = openShiftRequestMapping.KronosPersonNumber;
+
+                if (!approved)
+                {
+                    // Deny in Kronos, Update mapping for Teams.
+                    success = await this.openShiftRequestController.ApproveOpenShiftInKronos(kronosReqId, kronosUserId, openShiftRequestMapping, approved).ConfigureAwait(false);
+                    if (!success)
+                    {
+                        responseModelList.Add(GenerateResponse(openShiftRequest.Id, HttpStatusCode.BadRequest, null, null));
+                        return responseModelList;
+                    }
+
+                    responseModelList.Add(GenerateResponse(openShiftRequest.Id, HttpStatusCode.OK, null, null));
+                    openShiftRequestMapping.ShiftsStatus = ApiConstants.Refused;
+                    await this.openShiftRequestMappingEntityProvider.SaveOrUpdateOpenShiftRequestMappingEntityAsync(openShiftRequestMapping).ConfigureAwait(false);
+                    return responseModelList;
+                }
+
+                // approve in kronos
+                success = await this.openShiftRequestController.ApproveOpenShiftInKronos(kronosReqId, kronosUserId, openShiftRequestMapping, approved).ConfigureAwait(false);
+                responseModelList.Add(GenerateResponse(openShiftRequest.Id, HttpStatusCode.OK, null, null));
+
+                if (success)
+                {
+                    var shiftsTemp = await this.shiftController.GetShiftsForUser(kronosUserId, openShiftRequestMapping.PartitionKey).ConfigureAwait(false);
+                    var date = this.utility.UTCToKronosTimeZone(shift.SharedShift.StartDateTime, kronosTimeZone).ToString("d", CultureInfo.InvariantCulture);
+
+                    // confirm new shift exists on kronos
+                    var newKronosShift = shiftsTemp
+                        .Schedule.ScheduleItems.ScheduleShift
+                        .Where(x => x.Employee.FirstOrDefault().PersonNumber == kronosUserId)
+                        .FirstOrDefault(x => x.StartDate == date);
+
+                    if (newKronosShift != null)
+                    {
+                        var kronosUniqueId = this.utility.CreateUniqueId(shift, kronosTimeZone);

```

Comment:
I think this should be newKronosShift

Reply:
The CreateUniqueId method creates the Kronos unique ID from a given Teams shift